### PR TITLE
modify site to fetch json

### PR DESCRIPTION
**Pull request by poncem91**
Commit modifies DataServlet java file to return JSON data, modifies script JS file to fetch the JSON data and present it when sayHello() function is called.

### CODE REVIEW

**Review by swparr — APPROVED** [review]

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -13,20 +13,34 @@
...
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that returns Hello Message */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+    
+    private ArrayList<String> helloMessages;
```
Comment (peap) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: nit: check the indentation here

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -119,7 +119,13 @@ document.addEventListener('keydown', function(event) {
 
 /**  */
 function sayHello() {
-  fetch('/data').then(response => response.text()).then((helloMessage) => {
-    document.getElementById('hello-container').innerText = helloMessage;
+  fetch('/data').then(response => response.json()).then((helloMessage) => {
```
Comment (peap) [inline portfolio/src/main/webapp/script.js:122]: I found a bug :) 

image

**Review by peap — CHANGES_REQUESTED** [review]

File: portfolio/src/main/webapp/script.js (hunk 2 above)
Comment (poncem91) [reply portfolio/src/main/webapp/script.js:122]: Thanks for catching that! Definitely not what I intended. Will fix this

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (poncem91) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Will fix this, thanks for pointing this out!

File: portfolio/src/main/webapp/script.js (hunk 2 above)
Comment (poncem91) [reply portfolio/src/main/webapp/script.js:122]: Done! :)

Comment (peap) [reply portfolio/src/main/webapp/script.js:122]: The `Hello!` button is gone from the Mafe_jsonWalkthrough branch, and I don't see it in master, either. It's still hanging around in Mafe_FetchWalkthrough, though. Maybe it got lost in some merge commit?

Comment (poncem91) [reply portfolio/src/main/webapp/script.js:122]: I tried merging one of the previous branches and it did some weird things... I just reverted some of it and will be making a new pull request for the Mafe_jsonWalkthrough branch (the button should be there now) along with a pull request for a branch with the POST form. I’m hoping this brings back to where they should be

**Pull request merged**